flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/metrics/MetricStore.java

```diff
@@ -320,7 +320,8 @@ private TaskManagerMetricStore() {
 			this(new ConcurrentHashMap<>(), ConcurrentHashMap.newKeySet());
 		}
 
-		private TaskManagerMetricStore(Map<String, String> metrics, Set<String> garbageCollectorNames) {
+		@VisibleForTesting
+		protected TaskManagerMetricStore(Map<String, String> metrics, Set<String> garbageCollectorNames) {
```

Comment:
I would avoid `VisibleForTesting` shortcuts.
we can use available `MetricStore::add` instead like in `MetricStoreTest`